packages/components/search/src/search.js

```diff
@@ -0,0 +1,108 @@
+import { TSElement, unsafeCSS, html, customElementDefineHelper, helpers } from '@tradeshift/elements';
+import css from './search.css';
+
+import '@tradeshift/elements.icon';
+
+customElementDefineHelper(
+	'ts-search',
+	class extends TSElement {
+		static get styles() {
+			return [TSElement.styles, unsafeCSS(css)];
+		}
+
+		static get properties() {
+			return {
+				autofocus: { type: Boolean, reflect: true },
+				dir: { type: String, reflect: true },
+				focused: { type: Boolean, reflect: true },
+				idleTime: { type: Number, attribute: 'idle-time', reflect: true },
+				placeholder: { type: String, reflect: true },
+				value: { type: String, reflect: true }
+			};
+		}
+
+		constructor() {
+			super();
+			this.idleTime = 300;
+			this.placeholder = 'Search...';
+			this.value = '';
+			this.dispatchIdleEvent = helpers.debounceEvent(() => this.dispatchCustomEvent('idle', this.value), this.idleTime);
+		}
+
+		get direction() {
+			return this.dir ? this.dir : this.bodyDir;
+		}
+
+		get showCancel() {
+			return Boolean(this.value);
+		}
+
+		attributeChangedCallback(name, oldVal, newVal) {
+			super.attributeChangedCallback(name, oldVal, newVal);
+			switch (name) {
+				case 'idle-time':
+					this.dispatchIdleEvent = helpers.debounceEvent(() => this.dispatchCustomEvent('idle', this.value), newVal);
+					break;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a "default" clause to this "switch" statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ts.sv/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)